### ISSUE TITLE
E: Unable to locate package datacenter-gpu-manager

### ISSUE THREAD

**Issue by cyLi-Tiger** [NVIDIA/DCGM#46]
I follow the install instruction [here](https://docs.nvidia.com/datacenter/dcgm/latest/user-guide/getting-started.html#:~:text=Now%2C-,install%20DCGM,-%24%20sudo%20apt). But find an error saying that unable to find the package.

My commands are followed:
1. apt-key del 7fa2af80
2. distribution=$(. /etc/os-release;echo $ID$VERSION_ID | sed -e 's/\.//g')
3. wget https://developer.download.nvidia.com/compute/cuda/repos/$distribution/x86_64/cuda-keyring_1.0-1_all.deb
4. dpkg -i cuda-keyring_1.0-1_all.deb
5. apt-get update
6. apt-get install -y datacenter-gpu-manager

On step 6 I met error "E: Unable to locate package datacenter-gpu-manager"

I wonder how to fix this problem?

**Comment by nikkon-dev** [comment]
What is your `$distribution` value?

**Comment by cyLi-Tiger** [comment]
> 

ubuntu1804

**Comment by nikkon-dev** [comment]
Could you check if adding the repository manually would work for you?

```
wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/cuda-ubuntu1804.pin
sudo mv cuda-ubuntu1804.pin /etc/apt/preferences.d/cuda-repository-pin-600
sudo apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub
sudo add-apt-repository "deb https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/ /"
sudo apt-get update
```

**Comment by cyLi-Tiger** [comment]
While waiting for your reply, I found another way to install DCGM using the method introduced in triton [Model Analyzer's install.md](https://github.com/triton-inference-server/model_analyzer/blob/main/docs/install.md#:~:text=installing%20DCGM%20on%20Ubuntu%2020.04%20you).

Now I can use CLI dcgmi but I met another mistake, while I was running ` dcgmi discovery -l`to check my gpu. I got
`Error: unable to establish a connection to the specified host: localhost.
Error: Unable to connect to host engine. Host engine connection invalid/disconnected.`

I wonder how to fix that problem?

**Comment by cyLi-Tiger** [comment]
> Update: this works after I change the source in /etc/apt/sources.list into aliyun

**Comment by nikkon-dev** [comment]
@cyLi-Tiger,
I hope you have not used the same 2.0.13 version as in the [Model Analyzer's install.md](https://github.com/triton-inference-server/model_analyzer/blob/main/docs/install.md#:~:text=installing%20DCGM%20on%20Ubuntu%2020.04%20you). That's an old version. The latest version is 2.4.6
This approach has a drawback: You will not automatically get newer versions via apt.

As for the error, you have not started the nvidia-dcgm service, so the nv-hostengine process is not running. The dcgmi command is just a CLI for the nv-hostengine and needs its server part to operate.

**Comment by cyLi-Tiger** [comment]
> @cyLi-Tiger, I hope you have not used the same 2.0.13 version as in the [Model Analyzer's install.md](https://github.com/triton-inference-server/model_analyzer/blob/main/docs/install.md#:~:text=installing%20DCGM%20on%20Ubuntu%2020.04%20you). That's an old version. The latest version is 2.4.6 This approach has a drawback: You will not automatically get newer versions via apt.
> 
> As for the error, you have not started the nvidia-dcgm service, so the nv-hostengine process is not running. The dcgmi command is just a CLI for the nv-hostengine and needs its server part to operate.

I noticed that version issue and I have installed the latest one.

The nvidia-dcgm service starting command I found in [DCGM document](https://docs.nvidia.com/datacenter/dcgm/latest/user-guide/getting-started.html#:~:text=sudo%20systemctl%20%2D%2Dnow%20enable%20nvidia%2Ddcgm) needs systemctl to start, but I am using docker, it seems that docker don't support `systemctl` command, is there any other ways to start nvidia-dcgm service, such as `service nvidia-dcgm start`?

**Comment by cyLi-Tiger** [comment]
> @cyLi-Tiger, I hope you have not used the same 2.0.13 version as in the [Model Analyzer's install.md](https://github.com/triton-inference-server/model_analyzer/blob/main/docs/install.md#:~:text=installing%20DCGM%20on%20Ubuntu%2020.04%20you). That's an old version. The latest version is 2.4.6 This approach has a drawback: You will not automatically get newer versions via apt.
> 
> As for the error, you have not started the nvidia-dcgm service, so the nv-hostengine process is not running. The dcgmi command is just a CLI for the nv-hostengine and needs its server part to operate.

@nikkon-dev 
I noticed that version issue and I have installed the latest one.

The nvidia-dcgm service starting command I found in [DCGM document](https://docs.nvidia.com/datacenter/dcgm/latest/user-guide/getting-started.html#:~:text=sudo%20systemctl%20%2D%2Dnow%20enable%20nvidia%2Ddcgm) needs systemctl to start, but I am using docker, it seems that docker don't support systemctl command, is there any other ways to start nvidia-dcgm service, such as service nvidia-dcgm start?

**Comment by nikkon-dev** [comment]
For dockerized environment, you will have to start `nv-hostengine` manually.
Here is an example: [Dockerfile](https://github.com/NVIDIA/DCGM/blob/master/docker/Dockerfile.ubuntu20.04)
There is also a ready-made Docker image for DCGM: [nvidia/dcgm](https://hub.docker.com/r/nvidia/dcgm)

Please remember that there should not be two nv-hostengine instances accessing the same hardware. This means you should not start two docker containers with nv-hostengine running.

**Comment by cyLi-Tiger** [comment]
> For dockerized environment, you will have to start `nv-hostengine` manually. Here is an example: [Dockerfile](https://github.com/NVIDIA/DCGM/blob/master/docker/Dockerfile.ubuntu20.04) There is also a ready-made Docker image for DCGM: [nvidia/dcgm](https://hub.docker.com/r/nvidia/dcgm)
> 
> Please remember that there should not be two nv-hostengine instances accessing the same hardware. This means you should not start two docker containers with nv-hostengine running.

Thanks for your reply. I have successfully used DCGM to monitor my GPU refer to [dcgm_monitor in Model Analyzer](https://github.com/triton-inference-server/model_analyzer/tree/main/model_analyzer/monitor).

But I still have 3 question about DCGM, I noticed that there are field identifiers like memory utilization and gpu utilization. 
1. How these methods are calculated?
2. What if I want to monitor the running status of each core of GPU, is there any api for that?
3. Is there any way to monitor the highest GPU memory use during a time period? I only have a toy plan: collect multiple records during a time period and return the maximum among them.

<img width="274" alt="image" src="https://user-images.githubusercontent.com/37681002/182296341-3368a8cd-cfeb-4f07-be36-d292b780040b.png">
